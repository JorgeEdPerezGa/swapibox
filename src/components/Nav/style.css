.nav-header {
  display: flex;
  justify-content: space-between;
  background-color: #222;
  padding: 20px;
  color: white;
}

.nav-menu {
  display: flex;
}

.nav {
  margin: 25px;
  color: #FFFFFF;
  text-decoration-line: none;
}

.active {
  color: #abfeff;
}

.nav-home {
  color: #FFFFFF;
}
