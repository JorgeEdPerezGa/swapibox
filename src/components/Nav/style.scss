.nav-header {
  position: fixed;
  z-index: 2;
  font-weight: 300;
  display: flex;
  justify-content: space-between;
  background-color: #222;
  color: white;
  width: 100%;
}

.nav-menu {
  display: flex;
}

.nav {
  margin: 25px;
  color: #FFFFFF;
  text-decoration-line: none;
}

.active {
  color: #abfeff;
}

.nav-home {
  color: #FFFFFF;
}

// .favorite-nav {
//   margin-left: 15px;
// }

.count {
  color: white;
  margin: 0;
  padding: 25px 0 25px 30px;
}
