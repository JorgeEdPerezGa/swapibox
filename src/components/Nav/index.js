import React from 'react';
import { NavLink } from 'react-router-dom';
import './style.css';

const Nav = () => {
  return (
    <header className = 'nav-header'>
      <NavLink className = 'nav nav-home' to = '/'>SWAPIBOX</NavLink>
      <div className = 'nav-menu'>
        <NavLink className = 'nav' to = '/people'>PEOPLE</NavLink>
        <p>PLANETS</p>
        <p>VEHICLES</p>
      </div>
    </header>
  );
};

export default Nav;
