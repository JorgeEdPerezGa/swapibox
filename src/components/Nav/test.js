import React from 'react';
import Nav from './index';
import setupTest from '../../setupTests';
import { shallow } from 'enzyme';

describe('<Nav/>', () => {
  const wrapper = shallow(<Nav />);

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
