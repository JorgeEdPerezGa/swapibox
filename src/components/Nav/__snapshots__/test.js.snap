// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav/> should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav nav-home"
          to="/"
>
          SWAPIBOX
</NavLink>,
        <div
          className="nav-menu"
>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav"
                    to="/people"
          >
                    PEOPLE
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav"
                    to="/planets"
          >
                    PLANETS
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav"
                    to="/vehicles"
          >
                    VEHICLES
          </NavLink>
          <p
                    className="count"
          />
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav favorite-nav"
                    to="/favorites"
          >
                    FAVORITES
          </NavLink>
</div>,
      ],
      "className": "nav-header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "SWAPIBOX",
          "className": "nav nav-home",
          "to": "/",
        },
        "ref": null,
        "rendered": "SWAPIBOX",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav"
              to="/people"
>
              PEOPLE
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav"
              to="/planets"
>
              PLANETS
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav"
              to="/vehicles"
>
              VEHICLES
</NavLink>,
            <p
              className="count"
/>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav favorite-nav"
              to="/favorites"
>
              FAVORITES
</NavLink>,
          ],
          "className": "nav-menu",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "PEOPLE",
              "className": "nav",
              "to": "/people",
            },
            "ref": null,
            "rendered": "PEOPLE",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "PLANETS",
              "className": "nav",
              "to": "/planets",
            },
            "ref": null,
            "rendered": "PLANETS",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "VEHICLES",
              "className": "nav",
              "to": "/vehicles",
            },
            "ref": null,
            "rendered": "VEHICLES",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "count",
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "FAVORITES",
              "className": "nav favorite-nav",
              "to": "/favorites",
            },
            "ref": null,
            "rendered": "FAVORITES",
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav nav-home"
            to="/"
>
            SWAPIBOX
</NavLink>,
          <div
            className="nav-menu"
>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav"
                        to="/people"
            >
                        PEOPLE
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav"
                        to="/planets"
            >
                        PLANETS
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav"
                        to="/vehicles"
            >
                        VEHICLES
            </NavLink>
            <p
                        className="count"
            />
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav favorite-nav"
                        to="/favorites"
            >
                        FAVORITES
            </NavLink>
</div>,
        ],
        "className": "nav-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "SWAPIBOX",
            "className": "nav nav-home",
            "to": "/",
          },
          "ref": null,
          "rendered": "SWAPIBOX",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav"
                to="/people"
>
                PEOPLE
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav"
                to="/planets"
>
                PLANETS
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav"
                to="/vehicles"
>
                VEHICLES
</NavLink>,
              <p
                className="count"
/>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav favorite-nav"
                to="/favorites"
>
                FAVORITES
</NavLink>,
            ],
            "className": "nav-menu",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "PEOPLE",
                "className": "nav",
                "to": "/people",
              },
              "ref": null,
              "rendered": "PEOPLE",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "PLANETS",
                "className": "nav",
                "to": "/planets",
              },
              "ref": null,
              "rendered": "PLANETS",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "VEHICLES",
                "className": "nav",
                "to": "/vehicles",
              },
              "ref": null,
              "rendered": "VEHICLES",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "count",
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "FAVORITES",
                "className": "nav favorite-nav",
                "to": "/favorites",
              },
              "ref": null,
              "rendered": "FAVORITES",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
